src/olympia/amo/fields.py

```diff
@@ -38,7 +39,17 @@ def __call__(self, value):
 
 
 class HttpHttpsOnlyURLField(fields.URLField):
-    default_validators = [URLValidatorBackport(schemes=('http', 'https'))]
+    default_validators = [
+        URLValidatorBackport(schemes=('http', 'https')),
+        # Reject AMO URLs, see:
+        # https://github.com/mozilla/addons-server/issues/9012
+        RegexValidator(
+            regex=r'%s' % settings.SERVICES_DOMAIN,
```

Comment:
You're testing for `SITE_URL` in the tests but only doing `SERVICES_DOMAIN` here. Don't we need both?

Reply:
no, I meant `SERVICES_DOMAIN` as defined here: https://github.com/mozilla/addons-server/blob/5dfda3a8661112ab816d92c14beafbd83a3efa4f/settings.py#L46

But, I like this `DOMAIN` variable better!

Reply:
`settings.py` is only used for local development.  https://github.com/mozilla/addons-server/blob/master/src/olympia/lib/settings_base.py is the base settings file, and some settings are overridden per server in https://github.com/mozilla/addons-server/tree/master/src/olympia/conf